sceptre/config/graph.py

```diff
@@ -2,87 +2,113 @@
 
 """
 sceptre.config.graph
-This module implements a StackConfig graph, which stores a directed graph
-of a stack's dependencies.
+
+This module implements a StackGraph, which is represented as a directed
+acyclic graph of a Stack's dependencies.
 """
 
 import logging
 import networkx as nx
 from sceptre.exceptions import CircularDependenciesError
 
 
-class StackDependencyGraph(object):
+class StackGraph(object):
     """
-    A Directed Graph representing the relationship between Stack config
-    dependencies. Responsible for initalizing the graph object based on
-    a given inital stack config path.
+    A Directed Acyclic Graph representing the relationship between a Stack
+    and its dependencies. Responsible for initalising the graph based on a set
+    of Stacks.
     """
 
-    def __init__(self, dependency_map=None):
+    def __init__(self, stacks):
         """
-        Graph that is based on a given `dependency_map`.
+        Initialises a StackGraph based on a `set` of Stacks.
 
-        :param dependency_map: A dict containing a list of the dependencies for
-        a given stack
-        :type dict: dict
+        :param stacks: A set of Stacks.
+        :type stacks: set
         """
         self.logger = logging.getLogger(__name__)
         self.graph = nx.DiGraph()
-        if dependency_map is None:
-            dependency_map = {}
-        self.dependency_map = dependency_map
-        self._generate_graph()
+        self._generate_graph(stacks)
 
     def __repr__(self):
         return str(nx.convert.to_dict_of_lists(self.graph))
 
-    def _generate_graph(self):
+    def __iter__(self):
+        return self.graph.__iter__()
+
+    def count_dependencies(self, stack):
+        """
+        Returns the number of incoming edges a given Stack has in the
+        StackGraph. The number of incoming edge also represents the number
+        of Stacks that depend on the given Stack.
+        """
+        return self.graph.in_degree(stack)
+
+    def remove_stack(self, stack):
         """
-        Generates the graph for the initalized StackDependencyGraph object
+        Removes a Stack from the StackGraph. This operation will also remove
+        all adjecent edges that represent a 'depends on' relationship with
+        other Stacks.
         """
-        for stack, dependencies in self.dependency_map.items():
-            self._generate_edges(stack, dependencies)
+        return self.graph.remove_node(stack)
 
-    def _is_acyclic(self):
-        return nx.is_directed_acyclic_graph(self.graph)
+    def write(self):
```

Comment:
remove `write()`. Was useful for debugging but not suitable for release in it's current form. Will also break as `pydot` is not listed in requirements.